core/src/test/java/com/es/core/cart/HttpSessionCartServiceUnitTest.java

```diff
@@ -0,0 +1,93 @@
+package com.es.core.cart;
+
+import com.es.core.model.phone.CartItem;
+import com.es.core.model.phone.Phone;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpSessionCartServiceUnitTest {
```

Comment:
This test class only covers two recalculation methods of the service, please provide 100% code coverage

Reply:
fixed